api/v1/relay/relay.proto

```diff
@@ -0,0 +1,152 @@
+// Copyright 2020 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+import "google/protobuf/timestamp.proto";
+import public "flow/flow.proto";
+
+package relay;
+
+service HubbleRelay {
+    rpc GetEvents(GetEventsRequest) returns (stream GetEventsResponse) {}
+}
+
+// Here I didn't include "follow", "until", and "number". This request assumes follow,
+// and lets the client decide when to end the request, whether it's based on timestamp
+// or the number of responses received.
+message GetEventsRequest{
+    // EventType specifies which types of events to subscribe to. If unspecified, the
+    // request is subscribed to all the event types.
+    repeated EventType event_types = 1;
+    repeated EventFilter blacklist = 2;
+    repeated EventFilter whitelist = 3;
```

Comment:
How are conflicts between the whitelist and blacklist intended to be resolved?

Reply:
I assume this will be the same as for the observer service:

    // blacklist defines a list of filters which have to match for a flow to be
    // excluded from the result.
    // If multiple blacklist filters are specified, only one of them has to
    // match for a flow to be excluded.
    repeated flow.FlowFilter blacklist = 5;

    // whitelist defines a list of filters which have to match for a flow to be
    // included in the result.
    // If multiple whitelist filters are specified, only one of them has to
    // match for a flow to be included.
    // The whitelist and blacklist can both be specified. In such cases, the
    // set of the returned flows is the set difference `whitelist - blacklist`.
    // In other words, the result will contain all flows matched by the
    // whitelist that are not also simultaneously matched by the blacklist.

If not, that'd be a source of confusion.